src/tbp/monty/frameworks/experiments/monty_experiment.py

```diff
@@ -441,16 +452,14 @@ def init_monty_data_loggers(self, logging_config: dict[str, Any]) -> None:
         )
 
     def get_epoch_state(self):
-        mode = self.model.experiment_mode
-
-        if mode == "train":
+        if self.experiment_mode == ExperimentMode.TRAIN:
```

Comment:
It isn't critical to change this since it works with `==`, but Enum values should be compared by identity using `is`. The check will be faster since `is` only looks at whether the two variables reference the same thing. See https://docs.python.org/3/howto/enum.html#comparisons
```suggestion
        if self.experiment_mode is ExperimentMode.TRAIN:
```

Reply:
Yeah, there's a bunch of assignments, in tests especially.